### PR TITLE
Added required info about cart processors to add-cart-items.md

### PR DESCRIPTION
**Pull request by wirduzen-felix**
**Attention:** these changes are referencing add-cart-processor-collector.md which doesn't seem to be published, yet.

add-cart-items.md was missing information about cart processors. These are required for the line items to be persisted into the database and for them to be dispatched to the storefront. Furthermore this pull request replaces the "hardcoded" line item type with a constant.

### CODE REVIEW

File: guides/plugins/plugins/checkout/cart/add-cart-items.md [hunk 1]
```diff
@@ -163,5 +165,51 @@ Implementing the `LineItemFactoryInterface` will force you to also implement thr
...
+    {
+        $lineItems = $original->getLineItems()->filterFlatByType(ExampleHandler::TYPE);
+
+        foreach ($lineItems as $lineItem){
+            $toCalculate->add($lineItem);
+        }
+    }
+}
+```
+{% endcode %}
```
Comment (elkmod) [inline guides/plugins/plugins/checkout/cart/add-cart-items.md:196]: Could you add a whitespace before and after the codeblocks, please? :)

File: guides/plugins/plugins/checkout/cart/add-cart-items.md [hunk 2]
```diff
@@ -163,5 +165,51 @@ Implementing the `LineItemFactoryInterface` will force you to also implement thr
...
+As you can see, this processor takes an "original cart" as an input and adds all instances of our example type to a second cart, which will actually be persisted.
+
+Of course you can use processors to do much more than this. Have a look at [adding cart processors and collectors](./add-cart-processor-collecor.md).
+
+Now register this processor in your `services.xml` like this:
```
Comment (elkmod) [inline guides/plugins/plugins/checkout/cart/add-cart-items.md:201]: Same here.

**Review by elkmod — CHANGES_REQUESTED** [review]

File: guides/plugins/plugins/checkout/cart/add-cart-items.md [hunk 3]
```diff
@@ -163,5 +165,51 @@ Implementing the `LineItemFactoryInterface` will force you to also implement thr
...
+        foreach ($lineItems as $lineItem){
+            $toCalculate->add($lineItem);
+        }
+    }
+}
+```
+{% endcode %}
+As you can see, this processor takes an "original cart" as an input and adds all instances of our example type to a second cart, which will actually be persisted.
+
+Of course you can use processors to do much more than this. Have a look at [adding cart processors and collectors](./add-cart-processor-collecor.md).
```
Comment (elkmod) [inline guides/plugins/plugins/checkout/cart/add-cart-items.md:199]: The link contains a typo "collecor"

**Comment by elkmod** [comment]
Hi @wirduzen-felix - thank you for your contribution! I created a new PR #451 to add that guide to the navigation, however, I want to check with the team first - maybe there's a reason it was not published yet. Once we figured that out, we can merge your PR :)

**Comment by wirduzen-felix** [comment]
@elkmod Thank you very much :). I added the whitespaces and fixed the typo.

**Comment by elkmod** [comment]
Awesome - thanks for your help - PR is merged!

**Pull request merged**